packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -98,25 +99,83 @@ export const Block = Node.create<IBlock>({
     };
   },
 
-  // TODO: should we parse <li>, <ol>, <h1>, etc?
   parseHTML() {
     return [
+      // For all content copied from within the editor.
       {
         tag: "div",
+        getAttrs: (element) => {
+          if (typeof element === "string") {
+            return false;
+          }
+
+          if (element.getAttribute("node") === "block") {
```

Comment:
My proposal would be to use data- attributes for this, I think it's a best practice; https://www.w3schools.com/tags/att_data-.asp

i.e.: data-bn-type (`bn-type` indicates it's the blocknote type)